src/controllers/controller.js

```diff
@@ -0,0 +1,218 @@
+'use strict';
+
+const Canva = require('../views/canva.js');
+const ReversiModel = require('../models/reversi.js');
+const {
+  BLACK_COLOR,
+  WHITE_COLOR,
+  AVAILABLE_NUMBER,
+  EMPTY_NUMBER,
+  PASS_TURN,
+} = require('../utils/constants.js');
+const { withOnChangeListener, getColorsNumbersByColorName } = require('../utils/helpers');
+
+const myselfModeButton = document.getElementById('myself-mode');
+const computerModeButton = document.getElementById('computer-mode');
+
+class ReversiController extends Canva {
+  constructor(canvasContainer) {
+    super(canvasContainer);
+    this.IS_USER_MODE = true;
+    this.reversiModel = null;
+    this.matrix = [];
+    this.playerColor = withOnChangeListener('color', BLACK_COLOR, this.onPlayerChange);
+    this.whiteCounter = 2;
+    this.blackCounter = 2;
+    this.isFirstPlayerPassedTurn = false;
+    this.isClickBlocked = false;
+
+    this.generateStartBoard();
+  }
+
+  start() {
+    this.init();
+    this.renderBoard();
+    this.onModeChange(this.handleModeChange);
+    this.onBoardClick(this.handlePlayerClick);
+    this.reversiModel = new ReversiModel(this);
+    this.setAvailableSteps();
+  }
+
+  setAvailableSteps = () => {
+    this.cleanAvailableSteps();
+    const availableSteps = this.reversiModel.calculateAvailableSteps(this.matrix, this.playerColor.color);
+
+    if (availableSteps === PASS_TURN) {
+      if (!this.isFirstPlayerPassedTurn) {
+        this.isFirstPlayerPassedTurn = true;
+        this.togglePlayerColor();
+      } else {
+        this.handleGameOver();
+        return;
+      }
+    }
+
+    availableSteps.forEach(([row, column]) => this.matrix[row][column] = AVAILABLE_NUMBER);
+    this.renderBoard();
+    
+    if (!this.IS_USER_MODE && this.playerColor.color === WHITE_COLOR)
+      this.simulateAIMove(availableSteps);
+  };
+
+  generateStartBoard = () => {
+    // 0 corresponds for EMPTY cell, 1 - for WHITE and 2 - for BLACK, 3 - for AVAILABLE step
+    this.matrix = [
+      [0, 0, 0, 0, 0, 0, 0, 0],
+      [0, 0, 0, 0, 0, 0, 0, 0],
+      [0, 0, 0, 0, 0, 0, 0, 0],
+      [0, 0, 0, 1, 2, 0, 0, 0],
+      [0, 0, 0, 2, 1, 0, 0, 0],
+      [0, 0, 0, 0, 0, 0, 0, 0],
+      [0, 0, 0, 0, 0, 0, 0, 0],
+      [0, 0, 0, 0, 0, 0, 0, 0],
+    ];
```

Comment:
1. `Array(10).fill(Array(10).fill(0))`

2. And then place the starting position using arrays indexes